ProcessesApi/V1/Services/SoleToJointService.cs

```diff
@@ -102,13 +102,26 @@ private async Task OnProcessClosed(Stateless.StateMachine<string, string>.Transi
         {
             var processRequest = x.Parameters[0] as ProcessTrigger;
 
-            SoleToJointHelpers.ValidateFormData(processRequest.FormData, new List<string>() { SoleToJointFormDataKeys.HasNotifiedResident });
+            if (processRequest.FormData.ContainsKey(SoleToJointFormDataKeys.Reason))
+            {
+                SoleToJointHelpers.ValidateFormData(processRequest.FormData, new List<string>() { SoleToJointFormDataKeys.HasNotifiedResident, SoleToJointFormDataKeys.Reason });
+                var expectedFormDataKeys = new Dictionary<string, object>()
```

Comment:
This could be renamed? It doesn't seem to be expectedFormDataKeys.
Could also just do:
```csharp
 _eventData = ...
```